dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SampleReaderTest.java

```diff
@@ -0,0 +1,156 @@
+
+@RunWith(SpringRunner.class)
+@JdbcTest
+public class SampleReaderTest {
+
+    private static final String CHICKEN_ASSEMBLY_4 = "Gallus_gallus-4.0";
+
+    private static final String CHICKEN_ASSEMBLY_5 = "Gallus_gallus-5.0";
+
+    private static final String PRIMARY_ASSEMBLY = "Primary_Assembly";
+
+    private static final String NON_NUCLEAR = "non-nuclear";
+
+    private static final int PAGE_SIZE = 2000;
+
+    public static final int BATCH_ID = 12070;
+
+    public static final int WRONG_BATCH_ID = -1;
+
+    public static final int FIRST_SUBMITTED_INDIVIDUAL_ID = 6480;
+
+    public static final int SECOND_SUBMITTED_INDIVIDUAL_ID = 6483;
+
+    public static final String DBSNP_RELEASE = "150";
+
+    @Autowired
+    private DataSource dataSource;
+
+    private SampleReader reader;
+
+    private List<Sample> expectedSamples;
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setUp() throws Exception {
+        reader = buildReader(BATCH_ID, CHICKEN_ASSEMBLY_5, Collections.singletonList(PRIMARY_ASSEMBLY), PAGE_SIZE);
+
+        Map<String, String> cohorts = new HashMap<>();
+        cohorts.put(SampleRowMapper.POPULATION, "RBLS");
+
+        expectedSamples = new ArrayList<>();
+        String firstSampleId = buildSampleId(BATCH_ID, FIRST_SUBMITTED_INDIVIDUAL_ID);
+        expectedSamples.add(new Sample(firstSampleId, Sex.MALE, null, null, cohorts));
+        String secondSampleId = buildSampleId(BATCH_ID, SECOND_SUBMITTED_INDIVIDUAL_ID);
+        expectedSamples.add(new Sample(secondSampleId, Sex.UNKNOWN_SEX, null, null, cohorts));
+    }
+
+    private static String buildSampleId(int batchId, int submittedIndividualId) {
+        return String.valueOf(batchId) + "_" + String.valueOf(submittedIndividualId);
+    }
+
+    private SampleReader buildReader(int batch, String assembly, List<String> assemblyTypes, int pageSize)
+            throws Exception {
+        SampleReader fieldsReader = new SampleReader(DBSNP_RELEASE, batch, assembly, assemblyTypes, dataSource,
+                                                     pageSize);
+        fieldsReader.afterPropertiesSet();
+        ExecutionContext executionContext = new ExecutionContext();
+        fieldsReader.open(executionContext);
+        return fieldsReader;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        reader.close();
+    }
+
+    @Test
+    public void testLoadData() {
+        assertNotNull(reader);
+        assertEquals(PAGE_SIZE, reader.getPageSize());
+    }
+
+    @Test
+    public void testQuery() throws Exception {
+        List<Sample> list = readAll(reader);
+        assertEquals(2, list.size());
+        for (Sample sample : list) {
+            assertNotNull(sample);
+            assertTrue("Retrieved an unexpected sample: " + sample.toString(), expectedSamples.contains(sample));
+        }
+    }
+
+    private <T> List<T> readAll(ItemReader<T> reader) throws Exception {
```

Comment:
If this method is just used in one test, I will modify the test to use directly the reader, checking each read sample and the count at the end

Reply:
In fact there is a very similar method in another test class so maybe they could be merged?

Reply:
as always, I don't like utils classes, but there's no other place to put this kind of methods

Reply:
Parent class with protected method? It would make pretty easy to reuse it in every single reader test class.